platformio.ini

```diff
@@ -1144,18 +1144,25 @@ debug_tool      = jlink
 upload_protocol = jlink
 
 #
-# FLSUN QQ (STM32F103VET6)
+# FLSUN QQS Pro (STM32F103VET6) using hal STM32
+# board Hispeedv1
 #
-[env:flsun_hispeed]
-platform          = ${common_stm32f1.platform}
-extends           = common_stm32f1
-board             = genericSTM32F103VE
-extra_scripts     = ${common.extra_scripts}
-  buildroot/share/PlatformIO/scripts/mks_robin_mini.py
-  buildroot/share/PlatformIO/scripts/add_nanolib.py
-build_flags       = ${common_stm32f1.build_flags} -DMCU_STM32F103VE -DSS_TIMER=4
-lib_deps          = SoftwareSerialM
-  #Adafruit NeoPixel=https://github.com/Foxies-CSTL/Robin-NeoPixel-Lib/archive/master.zip
+[env:flsun_hispeedv1]
+platform             = ${common_stm32.platform}
+extends              = common_stm32
+build_flags          = ${common_stm32.build_flags} -DMCU_STM32F103VE -DSS_TIMER=4 -DENABLE_HWSERIAL3 -DTRANSFER_CLOCK_DIV=8
+board                = genericSTM32F103VE
+board_build.core     = stm32
+board_build.variant  = MARLIN_F103Vx
+board_build.ldscript = ldscript.ld
+board_build.offset   = 0x7000
+board_build.firmware = Robin_mini.bin
+board_upload.offset_address = 0x08007000
+build_unflags        = ${common_stm32.build_unflags} -DUSBCON -DUSBD_USE_CDC
+extra_scripts        = ${common.extra_scripts}
+  pre:buildroot/share/PlatformIO/scripts/generic_create_variant.py
+  buildroot/share/PlatformIO/scripts/stm32_bootloader.py
+  buildroot/share/PlatformIO/scripts/mks_encrypt.py
```

Comment:
@Foxies-CSTL is this board so identical to the MKS board that it even uses the same encrypt script?

Reply:
Yes, the first QQS came with the MKS_Robin_mini V2 but later for cost reasons (or bought the MKS license), they modified the electronics of the card (removal of E2 and of the PB2 connector, addition of a base for removable drivers, etc.). But they kept the compatibility with the Robin_mini.

Reply:
Hi @sjasonsmith on a random commit! I was just about to delete `add_nanolib.py` since it isn't used any more. But before I do, can you think of any reason that we would want to keep it?

Reply:
(It appears that `STM32F1/HAL/build_flags.py` takes care of adding `--specs=nano.specs` now.)